src/asm/x86/cdef.rs

```diff
@@ -58,59 +58,83 @@ pub(crate) unsafe fn cdef_filter_block<T: Pixel>(
       bit_depth,
       xdec,
       ydec,
+      edges,
       cpu,
     );
   };
-  #[cfg(feature = "check_asm")]
-  let ref_dst = {
-    let mut copy = dst.scratch_copy();
-    call_rust(&mut copy.as_region_mut());
-    copy
-  };
-  match T::type_enum() {
-    PixelType::U8 => {
-      match CDEF_FILTER_FNS[cpu.as_index()][decimate_index(xdec, ydec)] {
-        Some(func) => {
-          (func)(
-            dst.data_ptr_mut() as *mut _,
-            T::to_asm_stride(dst.plane_cfg.stride),
-            src,
-            src_stride,
-            pri_strength,
-            sec_strength,
-            dir as i32,
-            damping,
-          );
+
+  // TODO: handle padding in the fast path
+  if edges != CDEF_HAVE_ALL {
+    call_rust(dst);
+  } else {
+    #[cfg(feature = "check_asm")]
+    let ref_dst = {
+      let mut copy = dst.scratch_copy();
+      call_rust(&mut copy.as_region_mut());
+      copy
+    };
+    match T::type_enum() {
+      PixelType::U8 => {
+        match CDEF_FILTER_FNS[cpu.as_index()][decimate_index(xdec, ydec)] {
+          Some(func) => {
+            // current cdef_filter_block asm does 16->8 for historical
+            // reasons.  Copy into tmp space for now (also handling
+            // padding) until asm is updated
+            let tmpstride = 16; /* 128-bit alignment greater than 2 * (8>>xdec) + 2 */
```

Comment:
We should probably factor out all the alignment so bumping to avx512 or similar wide-vector-isa will not bite us.

Reply:
The alignment there is only to further a hack that had *better* be short-lived-- I don't want that copy/temp space at all.  All the dav1d asm handles the border cases internally, our current implementation doesn't.  So that's literally some duct-tape to keep ours running for right now.